modules/standard/Maps.chpl

```diff
@@ -0,0 +1,519 @@
+/*
+ * Copyright 2004-2019 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+  This module contains the implementation of the map type which is a container
+  that stores key-value associations. 
+
+  Maps are not parallel safe by default, but can be made parallel safe by
+  setting the param formal `parSafe` to true in any map constructor. When
+  constructed from another map, the new map will inherit the parallel safety
+  mode of its originating map.
+*/
+module Maps {
+
+  // Lock code lifted from modules/standard/Lists.chpl.
+  // Maybe they should be combined into a Locks module.
```

Comment:
I agree, but I think we'll need a bit more than just a simple spinlock before we make a user facing Lock module.